### PR TITLE
core[patch]: Fix stream events and stream log with global backgrounded callbacks, add warning

### PR DESCRIPTION
**Pull request by jacoblee93**
Makes certain callback handlers always awaited.

CC @nfcampos @hinthornw

### CODE REVIEW

File: langchain-core/src/callbacks/manager.ts [hunk 1]
```diff
@@ -20,6 +20,19 @@ import { consumeCallback } from "./promises.js";
...
 
+if (
+  getEnvironmentVariable("LANGCHAIN_TRACING_V2") === "true" &&
+  getEnvironmentVariable("LANGCHAIN_CALLBACKS_BACKGROUND") !== "true"
+) {
+  console.warn(
+    [
+      "[WARN]: You have enabled LangSmith tracing without backgrounding callbacks.",
+      "[WARN]: If you are not using a serverless environment where you must wait for tracing calls to finish,",
+      `[WARN]: we suggest setting "process.env.LANGCHAIN_CALLBACKS_BACKGROUND=true" to avoid additional latency.`,
```
Comment (hinthornw) [inline langchain-core/src/callbacks/manager.ts:31]: Makes sense. We may want to do this in the `traceable` HOF in the SDK as well

File: langchain-core/src/tracers/log_stream.ts [hunk 2]
```diff
@@ -320,7 +320,7 @@ export class LogStreamCallbackHandler extends BaseTracer {
   name = "log_stream_tracer";
 
   constructor(fields?: LogStreamCallbackHandlerInput) {
-    super(fields);
+    super({ _awaitHandler: true, ...fields });
```
Comment (hinthornw) [inline langchain-core/src/tracers/log_stream.ts:323]: oh cool

**Review by hinthornw — APPROVED** [review]

File: langchain-core/src/callbacks/manager.ts (hunk 1 above)
Comment (jacoblee93) [reply langchain-core/src/callbacks/manager.ts:31]: Yes let's do it

**Comment by jacoblee93** [comment]
With message history and run collector also did not work with this setting, fixes those too

**Pull request merged**